h/api/search.py

```diff
@@ -31,7 +34,41 @@ def _match_clause_for_uri(uristr):
     }
 
 
-def build_query(request_params):
+def scan(es_client, query, fields):
```

Comment:
Hmm, well the user just has to pass `request.es_client`, doesn't have to actually know anything about how Es works. But passing that through everything is boilerplatey. This makes me think that maybe the functions in `search.py` should be combined into an `EsClient` class, then `request.es_client` could become an instance of this class, then the code that wants to use it just does `request.es_client.scan(query, fields)`. (This is how `request.api_client works too). What do you think?